src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/StorePal.cs

```diff
@@ -283,7 +273,11 @@ private static void LoadMachineStores()
                 }
             }
 
-            s_machineRootStore = rootStore;
+            var rootStorePal = new CollectionBackedStoreProvider(rootStore, cloneOnRead: true);
+            s_machineIntermediateStore = new CollectionBackedStoreProvider(intermedStore, cloneOnRead: true);
+
+            // s_machineRootStore's nullarity is the loaded-state sentinel, so write it with Volatile.
+            Volatile.Write(ref s_machineRootStore, rootStorePal);
```

Comment:
Yeah, the very beginning of the method asserted that the monitor was held.  I can duplicate the assert to down here to make it more clear.

```C#
if (s_machineRootStore == null)
    lock (s_lock)
        if (s_machineRootStore == null)
            LoadMachineStores();  // <-- you are here
```